contracts/facets/ThirdPartyFacet.sol

```diff
@@ -189,9 +210,29 @@ contract ThirdPartyFacet {
         );
     }
 
+    /**
+     * @notice Transfers the value of the cursed bond of the archaeologist back to them, and un-curses their bond.
+     * @param sarcoId The identifier of the sarcophagus for which the bonds were cursed
+     * @param arch Address of the archaeologist to reimburse
+     */
+    function _reimburseArch(bytes32 sarcoId, address arch) private {
+        LibTypes.ArchaeologistStorage storage goodArch = s
+            .sarcophagusArchaeologists[sarcoId][arch];
+
+        uint256 cursedBond = LibBonds.calculateCursedBond(
+            goodArch.diggingFee,
+            goodArch.bounty
+        );
+
+        s.sarcoToken.transfer(arch, cursedBond);
```

Comment:
@DarksightKellar -- this line should be removed. You may be handling this in your current branch, so I will get this merged but wanted to point it out.